### ISSUE TITLE
Loading random theme from list of themes

### ISSUE THREAD

**Issue by Gnouc** [robbyrussell/oh-my-zsh#4647]
Currently, `oh-my-zsh` only allow user to load a specified theme or a random theme from all available theme.

From user perspective, it can make some inconvenience, because some themes setting is not available in user system by default. Example with `adben` theme, I always get:

```
ps1_fortune:2: command not found: fortune
```

It would be better if user can chose which set of themes they want to load:

```
ZSH_THEME=( robbyrussell fino agnoster )
```

will make `zsh` load one of three themes in each session.

If you guys agree with this feature, I will go with a pull request.

**Comment by apjanke** [comment]
Sounds useful. Instead of introducing an alternate list form of `$ZSH_THEME` itself, it might be easier in terms of implementation to add a separate variable like `$ZSH_THEME_RANDOM_CANDIDATES` that modifies the behavior of `ZSH_THEME=random`. Like #3703 does for implementing the converse of this, excluding particular themes from the random selection. #3743 also has some pending changes that add `random` and blacklist support to the `theme()` function that could incorporate this as well.

**Comment by Gnouc** [comment]
@apjanke Good point, implement a separate variable is better.

I make an attempt in this [commit](https://github.com/Gnouc/oh-my-zsh/commit/aac81241ecc86329c593ee7f0467c4d1e849e416).

**Comment by Gnouc** [comment]
@apjanke Would you please looking in the commit above :)

**Comment by apjanke** [comment]
I left a note on minor spelling and syntax. Otherwise looks good. Make it a PR and I'll test it out.

**Comment by robomc** [comment]
If anyone just wants to add this to their .zshrc in the meantime, add:

        ZSH_THEME_RANDOM_CANDIDATES=( jtriley simple sorin fino agnoster jispwoso tjkirch macovsky ys wezm theunraveler sunrise funky cloud bureau agnoster wezm+ )

        if [ "${(t)ZSH_THEME_RANDOM_CANDIDATES}" = "array" ] && [ "${#ZSH_THEME_RANDOM_CANDIDATES[@]}" -gt 0 ]; then
          themes=(${^ZSH_THEME_RANDOM_CANDIDATES})
          N=${#themes[@]}
          ((N=(RANDOM%N)+1))
          ZSH_THEME=${themes[$N]}
          echo $ZSH_THEME
        fi

        source $ZSH/oh-my-zsh.sh

**Comment by Gnouc** [comment]
@robomc I still use my branch in local to enable this feature, not sure when this will be pushed to the master.

Any ideal @apjanke?

**Comment by danhimalplanet** [comment]
Any progress on merging this to master?  This would be really useful for Linux stuff that doesn't work in OSX, for OSX stuff that doesn't work in Linux.

**Comment by ctf0** [comment]
https://github.com/robbyrussell/oh-my-zsh/issues/3271 fixed for me but the fortune shoul be optional depend if the package is installed or not, pushing the user to instal it is bad experiance.

**Comment by Gnouc** [comment]
@ctf0 We need to wait https://github.com/robbyrussell/oh-my-zsh/pull/3743 to be merged, no ETA at all

**Comment by apjanke** [comment]
>  the fortune shoul be optional depend if the package is installed or not, pushing the user to instal it is bad experiance.

Agreed. It should make the fortune element conditional. I'd consider that a bug in the individual theme using `fortune`. You can submit a separate PR to fix it, or we'll get around to it eventually.

**Comment by ctf0** [comment]
@Gnouc with the current huge number of PRs , and having the @apjanke fix made like a year a go , i wouldn't hold high hopes on that 😞

**Comment by nicolaisueper** [comment]
This isn't a topic anymore?

**Comment by robbyrussell** [comment]
Merged #4711 today!
